pkg/controller/logstorage/users/users_controller.go

```diff
@@ -357,15 +358,51 @@ func (r *UsersCleanupController) Reconcile(ctx context.Context, request reconcil
 		request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
 	reqLogger.Info("Reconciling LogStorage - Cleanup")
 
-	// Wait for Elasticsearch to be installed and available.
-	elasticsearch, err := utils.GetElasticsearch(ctx, r.client)
-	if err != nil {
+	// We skip requests without a namespace specified in multi-tenant setups.
+	if r.multiTenant && request.Namespace == "" {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if this is a tenant-scoped request.
+	_, _, err := utils.GetTenant(ctx, r.multiTenant, r.client, request.Namespace)
+	if errors.IsNotFound(err) {
+		reqLogger.Info("No Tenant in this Namespace, skip")
+		return reconcile.Result{}, nil
+	} else if err != nil {
 		return reconcile.Result{}, err
 	}
-	if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+
+	// Get LogStorage resource.
+	logStorage := &operatorv1.LogStorage{}
+	err = r.client.Get(ctx, utils.DefaultTSEEInstanceKey, logStorage)
+	if err != nil {
+		// Not finding the LogStorage CR is not an error, as a Managed cluster will not have this CR available but
+		// there are still "LogStorage" related items that need to be set up
```

Comment:
Thinks we need two SetDegraded calls in this branch - one for the error case and one for the NotFound case.

Reply:
Do we normally degrade if the CR isn't found? If this controller is tied to the CR and the CR doesn't exist then we should report any sort of status.

Reply:
Oh yeah, you're right. We would typically call OnCRNotFound instead I think.

Reply:
I wanted to avoid a new status since I do not know how we can track it since it refers to deleted tenants.